scrooge-generator-tests/src/test/resources/gold_file_output_scala/com/twitter/scrooge/test/gold/thriftscala/GoldService.scala

```diff
@@ -982,27 +982,25 @@ object GoldService extends _root_.com.twitter.finagle.thrift.GeneratedThriftServ
         }
       }
 
-    private def recordRequest(method: ThriftMethod): Unit = {
-      if (perEndpointStats) {
-        val methodStats = ThriftMethodStats((if (serviceName != "") stats.scope(serviceName) else stats).scope(method.name))
-        methodStats.requestsCounter.incr()
+    private def recordRequest(methodStats: Option[ThriftMethodStats]): Unit = {
+      methodStats.foreach { stats =>
```

Comment:
we like to avoid `Option.foreach` as it creates a `Function1`. instead could you use:
```
methodStats match {
  case Some(stats) => ...
  case None => ()
}
```